azure-pipelines.yml

```diff
@@ -1,342 +1,396 @@
+                  terraformInitSubscription: ${{ variables.tfInitSub }}
+                  product: ${{ variables.product }}
+                  tfVarsFile: "$(System.DefaultWorkingDirectory)/$(buildRepoSuffix)/environments/managed-identity/${{ component.env }}.tfvars"
+                  initCommandOptions: >
+                    -var subscription_id=${{ component.terraformSubscriptionID }}
+                    -backend-config resource_group_name=${{ component.backendAzureRmResourceGroupName }}
+                  planCommandOptions: >
+                    -var subscription_id=${{ component.terraformSubscriptionID }}
+                    -var control_vault=$(controlKeyVault)
+                    -var project=$(project)
 
-  - stage: Managed_Identity
-    displayName: 'Managed_Identity'
-    dependsOn: Network
-    jobs:
-      - job: Managed_Identity
-        steps:
-          - template: steps/terraform.yaml@cnp-azuredevops-libraries
-            parameters:
-              overrideAction: ${{ parameters.action }}
-              environment: ${{ parameters.env }}
-              component: managed-identity
-              location: ${{ parameters.location }}  
-              serviceConnection: $(serviceConnection)
-              terraformInitSubscription: ${{ variables.tfInitSub }}
-              product: ${{ variables.product }}
-              tfVarsFile: "$(System.DefaultWorkingDirectory)/$(buildRepoSuffix)/environments/managed-identity/${{ parameters.env }}.tfvars"
-              initCommandOptions: >
-                -var subscription_id=${{ variables.terraformSubscriptionID }}
-                -backend-config resource_group_name=$(backendAzureRmResourceGroupName)
-              planCommandOptions: >
-                -var subscription_id=${{ variables.terraformSubscriptionID }}
-                -var control_vault=$(controlKeyVault)
-                -var project=$(project)
+      - stage: "Aks_${{ component.env }}"
+        displayName: "${{ upper(component.env) }}: AKS clusters"
+        dependsOn:
+          - "CheckingClusters_${{ component.env }}"
+          - "Managed_Identity_${{ component.env }}"
+        variables:
+          existing_clusters: $[stageDependencies.CheckingClusters_${{ component.env }}.CheckingClusters.outputs['check_clusters.cluster_deploy']]
+          deploy_cluster: ${{ parameters.cluster }}
+        condition: |
+            or(
+              and(
+                eq(variables['Build.Reason'], 'Manual'),
+                in('${{ parameters.cluster }}', 'All', '00', '01'),
+                in(dependencies.CheckingClusters_${{ component.env }}.outputs['CheckingClusters.check_clusters.cluster_deploy'], 'All', '00', '01')
+              ),
+              and(
+                eq(dependencies.CheckingClusters_${{ component.env }}.outputs['CheckingClusters.check_clusters.cluster_deploy'], '${{ parameters.cluster }}'),
+                in('${{ parameters.cluster }}', '00', '01')
+              ),
```

Comment:
```
 and(
                eq(dependencies.CheckingClusters_${{ component.env }}.outputs['CheckingClusters.check_clusters.cluster_deploy'], '${{ parameters.cluster }}'),
                in('${{ parameters.cluster }}', '00', '01')
              ),
```

Isn't this condition the one where you also want to check the build reason is manual?

I think you can combine this and the one above into one check that sees:
if manual run, check cluster_deploy variable against parameter cluster if parameter cluster is not `All` ?

Still a little confusing to me unless you can explain what's going on here

Reply:
This checks if the output of the CheckingClusters task on the cluster_deploy output variable is equal to the cluster parameter. The second in('${{ parameters.cluster }}', '00', '01') checks both inputs on the cluster parameter

Reply:
Does that need to only happen when the build is manual too?